corehq/apps/app_manager/views/forms.py

```diff
@@ -348,6 +349,31 @@ def should_edit(attribute):
     if should_edit("shadow_parent"):
         form.shadow_parent_form_id = request.POST['shadow_parent']
 
+    if should_edit("custom_icon_form"):
+        text_body = request.POST.get("custom_icon_text_body")
+        xpath = request.POST.get("custom_icon_xpath")
+
+        # if there is a request to set custom icon
+        if request.POST.get("custom_icon_form"):
+            # validate that only of either text or xpath should be present
+            if (text_body and xpath) or (not text_body and not xpath):
+                return json_response(
+                    {'message': _("Please enter either text body or xpath for custom icon")},
+                    status_code=400
+                )
+            # a form should have just one custom icon for now
+            # so this just adds a new one with params or replaces the existing one with new params
+            form_custom_icon = (form.custom_icon if form.custom_icon else CustomIcon())
```

Comment:
nit: `form.custom_icon if form.custom_icon else CustomIcon()` this logic could go into `form.custom_icon` method itself, so you don't have to repeat this in multiple places and so that it will be easy to update this logic in one place latter if need be